File: InterconnectBackend/Controllers/Responses/BaseResponse.cs
namespace Models.Responses
{
    public class BaseResponse<TResponse>
    {
        public bool Success { get; set; }
        public string? ErrorMessage { get; set; }
        public TResponse? Data { get; set; }

        public static BaseResponse<TResponse> WithSuccess(TResponse data)
        {
            return new BaseResponse<TResponse>
            {
                Success = true,
                Data = data
            };
        }
        public static BaseResponse<object> WithEmptySuccess()
        {
            return new BaseResponse<object>
            {
                Success = true,
                Data = new object()
            };
        }
    }
}

File: InterconnectFrontend/src/api/responses/BaseResponse.ts
/**
 * Base response type for all API responses from the backend.
 * @template TData The type of data contained in the response
 */
interface BaseResponse<TData> {
    /** Whether the request was successful */
    success: boolean;
    /** Error message if the request failed, null if successful */
    errorMessage: string | null;
    /** Response data of generic type */
    data: TData;
}

export default BaseResponse;
